src/rinterface.c

```diff
@@ -7982,11 +7982,17 @@ SEXP R_igraph_hrg_sample(SEXP hrg) {
 
   SEXP r_result;
                                         /* Convert input */
-  R_SEXP_to_hrg(hrg, &c_hrg);
+  if (0 != R_SEXP_to_hrg_copy(hrg, &c_hrg)) {
+    igraph_error("", __FILE__, __LINE__, IGRAPH_ENOMEM);
+  }
+  IGRAPH_FINALLY(igraph_hrg_destroy, &c_hrg);
                                         /* Call igraph */
   IGRAPH_R_CHECK(igraph_hrg_sample(&c_hrg, &c_sample));
 
                                         /* Convert output */
+  PROTECT(hrg=R_igraph_hrg_to_SEXP(&c_hrg));
```

Comment:
This adds a `PROTECT()` without an `UNPROTECT()` . 

- Which logic is typically responsible for adding/counting `UNPROTECT()` calls?
- The converted `hrg` is never used. @szhorvat: is the `INOUT` signature for this argument in the `functions.yaml` correct?

Reply:
This must be resolved before we can merge this.